src/hooks/useLocalPaginator.ts

```diff
@@ -0,0 +1,83 @@
+import { useEffect, useState } from "react";
+
+export interface PageInfo {
+  endCursor: string;
+  hasNextPage: boolean;
+  hasPreviousPage: boolean;
+  startCursor: string;
+}
+
+export interface PaginationState {
+  after?: string;
+  before?: string;
+  first?: number;
+  last?: number;
+}
+
+export function useLocalPaginationState(
+  paginateBy: number
+): [PaginationState, (paginationState: PaginationState) => void] {
+  const [state, setState] = useState<PaginationState>({
+    first: paginateBy
+  });
+
+  const setPaginationState = (paginationState: PaginationState) => {
+    if (paginationState.after) {
+      setState({
+        after: paginationState.after,
+        first: paginateBy
+      });
+    } else if (paginationState.before) {
+      setState({
+        before: paginationState.before,
+        last: paginateBy
+      });
+    } else {
+      setState({
+        first: paginateBy
+      });
```

Comment:
I would also like to integrate it with `usePaginator` to keep this logic in one place even if callbacks are different (`setState` vs `navigate`)